classes/Tools.php

```diff
@@ -3058,12 +3058,13 @@ public static function getDirectories($path)
     public static function getDirectoriesWithGlob($path)
     {
         $directoryList = glob($path . '/*', GLOB_ONLYDIR | GLOB_NOSORT);
-        array_walk(
-            $directoryList,
-            function (&$absolutePath, $key) {
-                $absolutePath = substr($absolutePath, strrpos($absolutePath, '/') + 1);
-            }
-        );
+        if ($directoryList === false) {
+            $directoryList = [];
```

Comment:
we can early return the empty array 
```suggestion
            return [];
```

Reply:
I prefer to fix the fetch function result and keep then keep the compute path the same for all the cases.

Reply:
Why, if the array is empty, the array_map will do nothing :thinking:

Reply:
Why? If the array is empty, the array_map will do nothing :thinking:

Reply:
To handle everything with one path no matter if open_basedir is set or not. This is industry standard practice to prevent bugs comming from handling different usecases differently.

Reply:
sources ?

Reply:
yours? :D

Reply:
my source is a convention in the PrestaShop project. You'll see everywhere early returns. I didn't pull it off my hat

Reply:
https://github.com/PrestaShop/PrestaShop/pull/23969#issuecomment-829963771

Reply:
@sowbiba commited, hope you are happy 🥳